stumpy/core.py

```diff
@@ -1681,37 +1781,45 @@ def apply_exclusion_zone(a, idx, excl_zone, val):
     _apply_exclusion_zone(a, idx, excl_zone, val)
 
 
-def _preprocess(T):
+def _preprocess(T, copy=True):
     """
-    Creates a copy of the time series, transposes all dataframes, converts to
-    `numpy.ndarray`, and checks the `dtype`
+    Creates a copy of the time series when `copy` is True, transposes all dataframes,
+    converts to `numpy.ndarray`, and checks the `dtype`
 
     Parameters
     ----------
     T : numpy.ndarray
         Time series or sequence
 
+    copy : bool, default True
+        A boolean value that indicates whether the process should be done on
+        input `T` (False) or its copy (True).
+
     Returns
     -------
     T : numpy.ndarray
         Modified time series
     """
-    T = T.copy()
+    if copy:
+        T = T.copy()
     T = transpose_dataframe(T)
     T = np.asarray(T)
     check_dtype(T)
 
     return T
 
 
-def preprocess(T, m):
+def preprocess(T, m, copy=True, M_T=None, Σ_T=None, T_subseq_isconstant=None):
     """
     Creates a copy of the time series where all NaN and inf values
     are replaced with zero. Also computes mean and standard deviation
     for every subsequence. Every subsequence that contains at least
     one NaN or inf value, will have a mean of np.inf. For the standard
     deviation these values are ignored. If all values are illegal, the
-    standard deviation will be 0 (see `core.compute_mean_std`)
+    standard deviation will be 0 (see `core.compute_mean_std`). Also,
+    compute the rolling isconstant, a boolean array that indicates if
+    a subsequence is constant (True) or False. A subsequence is constant
+    if it only contains finite values that are all the same.
```

Comment:
"if it contains finite values that are IDENTICAL" 

Maybe it's just me but, subtly, I feel like "identical" seems stronger and less ambiguous?

Reply:
Agreed! I think the term "identical" can better deliver the message. I will revise it.